### ISSUE TITLE
[0.10.2] Reflections do not detect classes with Java 11.0.16 

### ISSUE THREAD

**Issue by cech12** [ronmamo/reflections#440]

**Comment by cech12** [comment]
So, it has something to do with this issue? https://github.com/ronmamo/reflections/issues/395

Then, a commit was made https://github.com/ronmamo/reflections/commit/7ccde6e7b6d8cc2e3f8fea4d7d3b3d2677214be2 and a merge is planned for 0.10.3.

**Comment by slovi** [comment]
Yes, It seems that commit 7ccde6e7b6d8cc2e3f8fea4d7d3b3d2677214be2 will fix also this issue.

**Comment by cech12** [comment]
@slovi Your workaround works for me! Thanks for that! :)
I hope, the 0.10.3 version will be released soon to remove that hack^^

**Comment by Lonzak** [comment]
I yesterday forked the project and tried the suggested bugfix https://github.com/ronmamo/reflections/commit/7ccde6e7b6d8cc2e3f8fea4d7d3b3d2677214be2 for issues (#395, #440, #427) However it did not help the error still occurs. Thus I completely removed the reflection lib and replaced the functionality with something else...

**Comment by michaelMarquard** [comment]
Maybe interesting for you to solve the issue:

I had a similar problem 
``java.lang.ClassCastException: class org.jboss.vfs.VirtualJarInputStream cannot be cast to class org.jboss.vfs.VirtualFile``
in the context of ``Java 11.0.20`` and ``WildFly Core 10.1.18.Final-redhat-00001`` (in a ``Redhat EAP 7.3.6 GA``).
Instead of 
``VirtualFile virtualFile = (VirtualFile) url.openConnection().getContent();``
I used 
``VirtualFile virtualFile = org.jboss.vfs.VFS.getChild(url.toURI());``
(no cast necessary) 
and it seems to do the job...

In ``Redhat EAP 7.4.11 GA`` (``WildFly Core 15.0.26.Final-redhat-00001``) the ``ClassCastException`` **does not** occur 
(there, ``url.openConnection().getContent()`` **is a** ``VirtualFile``; 
however, ``org.jboss.vfs.VFS.getChild(url.toURI())`` still works there as well).
